src/http-utils.js

```diff
@@ -117,10 +122,89 @@ function calculateRate(elapsed, totalTransferred) {
     return 0;
 }
 
+
+/**
+ * Validates and retrieves basic authentication information from a set of options. Will throw an error
+ * if only one of username or password is provided.
+ * @param {object} options Options from which "username" and "password" properties will be retrieved.
+ */
+function getBasicAuth(options) {
+    const { username, password } = options;
+    if (username && !password) {
+        throw new UploadError('password is required for basic auth', ErrorCodes.INVALID_OPTIONS);
+    }
+    if (password && !username) {
+        throw new UploadError('username is required for basic auth', ErrorCodes.INVALID_OPTIONS);
+    }
+    return options;
```

Comment:
What if both username and password are empty, or null or undefined?

Reply:
They're optional values, so they'll be ignored if they're _both_ missing. If one is specified, then the other must be also, but it's ok for both to be missing.